### PR TITLE
Fix/warnings

### PR DESCRIPTION
**Pull request by makosblade**
This PR:
- Fixes a bunch of mechanical warnings
- Removing unused imports
- Removing unnecessary boxing/unboxing
- Uses singletonList for one-element lists
- Uses Java classes instead of Guava for functional constructs
- Fixes compiler warnings for upgrading from Java 5 through Java 8

### CODE REVIEW

**Comment by makosblade** [comment]
Depends on https://github.com/geekbeast/rhizome-client/pull/32

File: src/main/java/com/openlattice/aws/AwsS3Pod.java [hunk 1]
```diff
@@ -52,7 +53,7 @@ public AmazonS3 awsS3() {
 
     public static AmazonS3 newS3Client( AmazonLaunchConfiguration awsConfig ) {
         AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
-        builder.setRegion( Region.getRegion( awsConfig.getRegion().or( Regions.DEFAULT_REGION ) ).getName() );
+        builder.setRegion( Region.getRegion( awsConfig.getRegion().orElse( Regions.DEFAULT_REGION ) ).getName() );
```
Comment (geekbeast) [inline src/main/java/com/openlattice/aws/AwsS3Pod.java]: Getting a compile error on this line.

Comment (makosblade) [reply src/main/java/com/openlattice/aws/AwsS3Pod.java]: Depends on https://github.com/geekbeast/rhizome-client/pull/32

Comment (makosblade) [reply src/main/java/com/openlattice/aws/AwsS3Pod.java]: So whenever the build artifacts decide to propagate. . . . . 🤔

Comment (geekbeast) [reply src/main/java/com/openlattice/aws/AwsS3Pod.java]: Happy Thanksgiving :)

**Pull request merged**